gatsby/create-node/create-preview-cards.js

```diff
@@ -1,65 +1,44 @@
-const slugify = require('slugify');
-const path = require('path');
-const { siteMetadata } = require('../../gatsby-config');
 const {
-  generateCard,
+  generateCardToBuffer,
 } = require('../util/preview-card-generator/generate-card');
+const { createFileNodeFromBuffer } = require('gatsby-source-filesystem');
 
-/**
- * Create a preview card file and put the url to the Gatsby store to be able
- * to link it in the page header.
- */
-const createPreviewMarkdown = ({ node, _, actions }) => {
-  const { createNodeField } = actions;
-  const post = node.frontmatter;
+const createPreviewCard = async (
+  title,
+  { node, _, actions, getCache, createNodeId },
+) => {
+  const { createNode, createNodeField } = actions;
 
   // we need the title in order to generate anything
-  if (!post.title) {
+  if (!title) {
     return;
   }
 
-  // we create a file name from the path (which is the page slug and more unique) but if none is available take the title
-  const fileName = `social-card---${slugify(post.path || post.title, {
-    lower: true,
-  })}.jpg`;
-  const outputFile = path.join('public', fileName);
-  const publicUrl = `${siteMetadata.siteUrl}/${fileName}`;
-
-  generateCard({ title: post.title }, outputFile).then(() => {
-    createNodeField({
-      node,
-      name: `socialCard`,
-      value: publicUrl,
-    });
+  const buffer = await generateCardToBuffer({ title });
+  const fileNode = await createFileNodeFromBuffer({
+    name: 'social-card',
+    buffer,
+    getCache,
+    createNode,
+    createNodeId,
   });
-};
 
-const createPreviewJob = ({ node, _, actions }) => {
-  const { createNodeField } = actions;
-
-  const fileName = `social-card---career-${slugify(node.name, {
-    lower: true,
-  })}.jpg`;
-
-  const outputFile = path.join('public', fileName);
-  const publicUrl = `${siteMetadata.siteUrl}/${fileName}`;
-
-  generateCard({ title: node.name }, outputFile).then(() => {
+  if (fileNode) {
     createNodeField({
       node,
-      name: 'socialCard',
-      value: publicUrl,
+      name: `socialCard`,
+      value: fileNode.id,
     });
-  });
+  }
 };
 
-const createPreviewCards = ({ node, _, actions }) => {
+const createPreviewCards = ({ node, ...rest }) => {
   if (node.internal.type === 'SyPersonioJob') {
-    createPreviewJob({ node, _, actions });
+    return createPreviewCard(node.name, { node, ...rest });
   }
 
   if (node.internal.type === 'MarkdownRemark') {
-    createPreviewMarkdown({ node, _, actions });
+    return createPreviewCard(node.frontmatter.title, { node, ...rest });
```

Comment:
I only need to pass in the actual string to generate. Before it was two methods while it's actually a single one 👍